app/controllers/course/assessment/assessments_controller.rb

```diff
@@ -71,10 +70,25 @@ def question_order_params
   end
 
   def assessment_params
-    params.require(:assessment).permit(:title, :description, :base_exp, :time_bonus_exp,
-                                       :start_at, :end_at, :bonus_end_at,
-                                       :published, :mode, :autograded, :password, :tabbed_view,
-                                       :skippable, :delayed_grade_publication, folder_params)
+    base_params = [:title, :description, :base_exp, :time_bonus_exp, :start_at, :end_at,
+                   :bonus_end_at, :published, :autograded]
+    if autograded?
+      base_params << :skippable
+    else
+      base_params += [:password, :tabbed_view, :delayed_grade_publication]
+    end
+    params.require(:assessment).permit(*base_params, folder_params)
+  end
+
+  # Infer the autograded state from @assessment or params.
+  def autograded?
+    if @assessment && @assessment.autograded
+      true
+    elsif @assessment && @assessment.autograded == false
```

Comment:
nope because `@assessment.autograded` could be `nil`, in that case it should go to `else`